platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -562,6 +563,12 @@ def _read_eeprom(self, offset, num_bytes, log_on_error=True):
                             # Indicate read finished
                             num_bytes = 0
                     if ctypes.get_errno() != 0:
+                        if retry < max_retry:
```

Comment:
not sure ctypes.get_errno works, maybe we should do the retry inside `except (OSError, IOError) as e`. Here is official definition of OSError:
```
class OSError(Exception):
    errno: int
    strerror: str
    # filename, filename2 are actually str | bytes | None
    filename: Any
    filename2: Any
    if sys.platform == "win32":
        winerror: int
```

We can use OSError.errno.

Reply:
ok. let's handle both. any concern if we do not distinguish error no# and retry on all errors?